// remotion/dist/cjs/interactivity-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sequenceSchemaDefaultLayoutNone = exports.sequenceSchemaWithoutFrom = exports.baseSchemaWithoutFrom = exports.sequenceSchema = exports.baseSchema = exports.freezeField = exports.fromField = exports.durationInFramesField = exports.extendSchemaWithSequenceName = exports.sequenceNameField = exports.hiddenField = exports.sequenceStyleSchema = exports.sequencePremountSchema = exports.premountSchema = exports.sequenceVisualStyleSchema = exports.transformSchema = void 0;
exports.transformSchema = {
    'style.transformOrigin': {
        type: 'transform-origin',
        step: 1,
        default: '50% 50%',
        description: 'Transform origin',
    },
    'style.translate': {
        type: 'translate',
        step: 1,
        default: '0px 0px',
        description: 'Offset',
    },
    'style.scale': {
        type: 'scale',
        max: 100,
        step: 0.01,
        default: 1,
        description: 'Scale',
    },
    'style.rotate': {
        type: 'rotation-css',
        step: 1,
        default: '0deg',
        description: 'Rotation',
    },
    'style.opacity': {
        type: 'number',
        min: 0,
        max: 1,
        step: 0.01,
        default: 1,
        description: 'Opacity',
        hiddenFromList: false,
    },
};
exports.sequenceVisualStyleSchema = exports.transformSchema;
exports.premountSchema = {
    premountFor: {
        type: 'number',
        default: 0,
        description: 'Premount For',
        min: 0,
        step: 1,
        hiddenFromList: false,
    },
    postmountFor: {
        type: 'number',
        default: 0,
        min: 0,
        step: 1,
        hiddenFromList: true,
    },
    styleWhilePremounted: {
        type: 'hidden',
    },
    styleWhilePostmounted: {
        type: 'hidden',
    },
};
exports.sequencePremountSchema = exports.premountSchema;
exports.sequenceStyleSchema = {
    ...exports.transformSchema,
    ...exports.premountSchema,
};
exports.hiddenField = {
    type: 'boolean',
    default: false,
    description: 'Hidden',
};
const showInTimelineField = {
    type: 'hidden',
};
exports.sequenceNameField = {
    type: 'hidden',
};
const extendSchemaWithSequenceName = (schema) => {
    return {
        name: exports.sequenceNameField,
        ...schema,
    };
};
exports.extendSchemaWithSequenceName = extendSchemaWithSequenceName;
exports.durationInFramesField = {
    type: 'number',
    default: undefined,
    min: 1,
    step: 1,
    hiddenFromList: true,
};
exports.fromField = {
    type: 'number',
    default: 0,
    step: 1,
    hiddenFromList: true,
};
exports.freezeField = {
    type: 'number',
    default: null,
    step: 1,
    hiddenFromList: true,
};
exports.baseSchema = {
    durationInFrames: exports.durationInFramesField,
    from: exports.fromField,
    freeze: exports.freezeField,
    hidden: exports.hiddenField,
    name: exports.sequenceNameField,
    showInTimeline: showInTimelineField,
};
exports.sequenceSchema = {
    ...exports.baseSchema,
    layout: {
        type: 'enum',
        default: 'absolute-fill',
        description: 'Layout',
        variants: {
            'absolute-fill': exports.sequenceStyleSchema,
            none: {},
        },
    },
};
exports.baseSchemaWithoutFrom = {
    durationInFrames: exports.durationInFramesField,
    freeze: exports.freezeField,
    hidden: exports.hiddenField,
    name: exports.sequenceNameField,
    showInTimeline: showInTimelineField,
};
exports.sequenceSchemaWithoutFrom = {
    ...exports.baseSchemaWithoutFrom,
    layout: exports.sequenceSchema.layout,
};
exports.sequenceSchemaDefaultLayoutNone = {
    ...exports.sequenceSchema,
    layout: {
        ...exports.sequenceSchema.layout,
        default: 'none',
    },
};
